Forge/src/main/java/me/ferdz/placeableitems/block/component/impl/BlazePowderBlockComponent.java

```diff
@@ -0,0 +1,33 @@
+package me.ferdz.placeableitems.block.component.impl;
+
+import me.ferdz.placeableitems.block.component.AbstractBlockComponent;
+import net.minecraft.block.BlockState;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+public class BlazePowderBlockComponent extends AbstractBlockComponent {
+    @Override
+    public void animateTick(BlockState stateIn, World worldIn, BlockPos pos, Random random) {
+        final double xSpreadRadius = 0.15, ySpreadRadius = 0.05;
+        worldIn.addParticle(
+                ParticleTypes.FLAME,
+                pos.getX() + 0.5 + randomFireSpread(xSpreadRadius, random),
+                pos.getY() + 0.2 + randomFireSpread(ySpreadRadius, random),
+                pos.getZ() + 0.5 + randomFireSpread(xSpreadRadius, random),
+                0,
+                0,
+                0
```

Comment:
Could you change it so the spread doesn't go out of the bounds of the block? Sometimes they appear on the blocks next to it. It appears that since you're using + for the spread radius, it also has a tendency to appear towards the high X of the block rather than the middle

Also maybe reduce the amount of particles to half using the `random`.

Reply:
Particles adjusted, I couldn't reproduce the tendency tho, it picks a random float between -spread and spread